### PR TITLE
Ab#54500 google re captcha

### PR DESCRIPTION
**Pull request by tropical-turtle**
Added Google reCatcha

### CODE REVIEW

File: src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml [hunk 1]
```diff
@@ -27,11 +27,11 @@
     <title>@WebTemplateModel.Render.HeaderTitle()</title>
     <meta content="width=device-width,initial-scale=1" name="viewport" />
     <!-- Load closure template scripts -->
-    <script type="text/javascript" src="@string.Concat(WebTemplateModel.CDNPath, "soyutils.js")"></script>
+    <script type="text/javascript" src="@string.Concat(WebTemplateModel.CDNPath, " soyutils.js")"></script>
     <link href="~/css/aurora.min.css" rel="stylesheet" />
     <link href="~/css/bootstrap/bootstrap.min.css" rel="stylesheet" />
     <link href="~/css/ErrorMessage.css" rel="stylesheet" />
-    <script type="text/javascript" src="@string.Format(System.Globalization.CultureInfo.InvariantCulture, "{0}wet-{1}.js", WebTemplateModel.CDNPath, WebTemplateModel.TwoLetterCultureLanguage)"></script>
+    <script type="text/javascript" src="@string.Format(System.Globalization.CultureInfo.InvariantCulture, " {0}wet-{1}.js", WebTemplateModel.CDNPath, WebTemplateModel.TwoLetterCultureLanguage)"></script>
```
Comment (AlexandreSanscartier) [inline src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml:34]: Remove space
```suggestion
    <script type="text/javascript" src="@string.Format(System.Globalization.CultureInfo.InvariantCulture, "{0}wet-{1}.js", WebTemplateModel.CDNPath, WebTemplateModel.TwoLetterCultureLanguage)"></script>
```

**Review by AlexandreSanscartier — CHANGES_REQUESTED** [review]

File: src/CSF.Web/CSF.Web.Client/Models/PageModels/UploadDocumentPageModel.cs [hunk 2]
```diff
@@ -49,11 +50,17 @@ public class UploadDocumentPageModel
...
-      
+        /// <summary>
         /// Gets or sets Renewing.
         /// </summary>
         public SubmissionType SubmissionType { get; set; }
 
+        /// <summary>
+        /// Gets or sets NotRobot value which is associated with recaptcha element.
+        /// </summary>
+        [Required(ErrorMessage ="\"I'm not a robot\" field must be checked")]
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.Web/CSF.Web.Client/Models/PageModels/UploadDocumentPageModel.cs]: Good point. Moved messaged to UploadDocumentValidator.cs and to the corresponding resource file.

File: src/CSF.Web/CSF.Web.Client/Pages/UploadDocument.razor [hunk 3]
```diff
@@ -26,6 +26,7 @@
 
 <EditForm Model="@model" OnValidSubmit="HandleValidSubmit">
     <FluentValidationValidator />
+    <DataAnnotationsValidator/>
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.Web/CSF.Web.Client/Pages/UploadDocument.razor]: Removed.

File: src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml (hunk 1 above)
Comment (tropical-turtle) [reply src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml]: Good catch.Space removed. Not sure how that space got introduced.

File: src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml [hunk 4]
```diff
@@ -43,6 +43,7 @@
     </script>
     @WebTemplateModel.Render.HtmlHeaderElements()
     <link href="~/css/main.css" rel="stylesheet" />
+    <script src="https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit" async defer></script>
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml]: Moved.

File: src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml [hunk 5]
```diff
@@ -27,11 +27,11 @@
     <title>@WebTemplateModel.Render.HeaderTitle()</title>
     <meta content="width=device-width,initial-scale=1" name="viewport" />
     <!-- Load closure template scripts -->
-    <script type="text/javascript" src="@string.Concat(WebTemplateModel.CDNPath, "soyutils.js")"></script>
+    <script type="text/javascript" src="@string.Concat(WebTemplateModel.CDNPath, " soyutils.js")"></script>
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml]: Removed the space.

File: src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml [hunk 6]
```diff
@@ -96,5 +96,6 @@
     <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
     <script src="~/js/Ninja.js"></script>
     <script src="~/js/aurora.min.js"></script>
+    <script src="https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit" async defer></script>
```
Comment (AlexandreSanscartier) [inline src/CSF.Web/CSF.Web.Client/Pages/_Host.cshtml:99]:
```suggestion
<script src="@string.Concat("https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit&hl=", @WebTemplateModel.TwoLetterCultureLanguage), " async defer"></script>
```

**Review by AlexandreSanscartier — CHANGES_REQUESTED** [review]
Added suggestion to localize the recaptcha.

**Review by AlexandreSanscartier — APPROVED** [review]

**Pull request merged**